src/lexer/scan.ts

```diff
@@ -233,9 +233,13 @@ export function scanSingleToken(parser: ParserState, context: Context, state: Le
             if (ch === Chars.Exclamation) {
               // Treat HTML begin-comment as comment-till-end-of-line.
               if (
-                parser.source.charCodeAt(parser.index + 2) === Chars.Hyphen &&
-                parser.source.charCodeAt(parser.index + 1) === Chars.Hyphen
+                parser.source.charCodeAt(parser.index + 1) === Chars.Hyphen &&
+                parser.source.charCodeAt(parser.index + 2) === Chars.Hyphen
               ) {
+                // Skip !--
+                advanceChar(parser);
```

Comment:
This could be tweaked to avoid member access and multiple `parser.source.charCodeAt` calls. E.g  `parser.source.charCodeAt(parser.index +3)`

Reply:
Not sure I follow, I want to make `currentChar` point to the beginning of the comment value so skipSingleLineComment will read the value correctly.

Reply:
This should work the same way `parser.currentChar = parser.source.charCodeAt(parser.index +3)` minus the extra `parser.source.charCodeAt` you use when using `advanceChar`

Reply:
Yeah but you're already reading charCodeAt when testing for ```parser.source.charCodeAt(parser.index + 2) === Chars.Hyphen &&
parser.source.charCodeAt(parser.index + 1) === Chars.Hyphen```
I suggest we convert that code to use advanceChar twice since we really do want to advance column index and check the current char.
The alternative would be to support something like:
```
export function advanceChar(parser: ParserState, advanceAmount: number = 1): number {
  parser.column += advanceAmount;
  return (parser.currentChar = parser.source.charCodeAt(parser.index += advanceAmount));
}
```
and call advanceChar(parser, 3)

Reply:
Agree there, so it's better to convert the 2 first lines to use advanceChar.